### PR TITLE
GitHub Action for Traffic Router unit tests

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

This PR
- [x] is not related to any Issue 
- adds a GitHub Action to run the Traffic Router unit tests
- Updates the Dockerized Traffic Router tests in `/traffic_router/tests` to use Java SE 8 instead of Java SE 6
- Makes `/traffic_router/tests` run on Alpine Linux by default but leaves a Dockerfile for CentOS 7 too in case anyone wants to run the tests on CentOS
- Adds optional debugging functionality to `/traffic_router/tests` and documents it

## Which Traffic Control components are affected by this PR?

- CI tests
- Traffic Router (Docker test runner)

## What is the best way to verify this PR?

* Check the result of the *Traffic Router Unit Tests* GitHub Action

    See the documentation in the *tr-tests* GitHub Action for instructions for using that Docker image to run the tests locally.
* Run the Traffic Router unit tests from the Docker unit test runner at `/traffic_router/tests`, make sure they pass
* Optional: Set breakpoints in the Traffic Router unit tests, then debug them referring to the given documentation

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: .github/workflows/tr.unit.tests.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Run unit tests
+        uses: ./.github/actions/tr-tests
+      - name: Upload Surefire Reports
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: surefire-reports
+          path: ${{ github.workspace }}/traffic_router/core/target/surefire-reports/*
```
Comment (ocket8888) [inline .github/workflows/tr.unit.tests.yaml:50]: Does it output GHA Annotations or are surefire reports the only rich view of failures we're gonna get?

File: traffic_router/tests/docker-compose.yml [hunk 2]
```diff
@@ -25,7 +25,13 @@ services:
   unit_router:
     build:
       context: ../..
-      dockerfile: traffic_router/tests/Dockerfile-test
+      dockerfile: traffic_router/tests/Dockerfile-test-alpine
+      #dockerfile: traffic_router/tests/Dockerfile-test-centos
```
[the remark this answers is not in the document]
Comment (zrhoffman) [reply traffic_router/tests/docker-compose.yml:29]: I kept it there for anyone who wants to use the CentOS Dockerfile, just so they know that it's there.

File: .github/workflows/tr.unit.tests.yaml (hunk 1 above)
Comment (zrhoffman) [reply .github/workflows/tr.unit.tests.yaml]: Okay, outputting GHA annotations in 95da3063bd.

**Review by ocket8888 — APPROVED** [review]
The action looks like it works fine, but I'm gonna ask someone to take a look at the TR Docker test stuff because I've never run those and probably won't with any great frequency in the future. So it'd be good to get the opinion of someone it'll actually affect.

**Pull request merged**